### PR TITLE
ExecutionPoker

### LINKED ISSUE #69: cli tool to play with Enforcer/Verifier - ExecutionPoker
# Bounty

## Deliverables
Introduce a command-line tool which interacts with Enforcer & Verifier,
as defined for Milestone `Song of Dawn`

## Gain for the project
Climbing up the roadmap 🗻 

## Roles
bounty size: 750 DAI
bounty gardener: @pinkiebell / 10%
bounty worker: @pinkiebell  / 70%
bounty reviewer: @johannbarbie  / 20%

### PR DESCRIPTION
**Pull request by pinkiebell**
🎉 command line interface for ExecutionPoker

Fix OP_STOP programCounter increment

Fixes #69

### CODE REVIEW

**Comment by johannbarbie** [comment]
how can i run the cli? don't see any instructions.

File: contracts/EVMRuntime.sol [hunk 1]
```diff
@@ -750,7 +750,7 @@ contract EVMRuntime is EVMConstants {
                 evm.pc = pc;
                 opcodeHandler(evm);
                 pcNext = evm.pc;
-            } else if (opcode == OP_RETURN || opcode == OP_REVERT || opcode == OP_STOP || opcode == OP_SELFDESTRUCT) {
+            } else if (opcode == OP_RETURN || opcode == OP_REVERT || opcode == OP_SELFDESTRUCT) {
```
Comment (johannbarbie) [inline contracts/EVMRuntime.sol]: why has OP_STOP been removed here?

File: contracts/mocks/SpendingConditionMock.sol [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+        for (uint i = 0; i < receivers.length; i++) {
+            token.transfer(receivers[i], amounts[i]);
+            token.balanceOf(receivers[i]);
+            // transferFrom(from, to, tokenid)
+            nft.balanceOf(receivers[i]);
+        }
+    }
+
+    function fulfil(
```
Comment (johannbarbie) [inline contracts/mocks/SpendingConditionMock.sol]: this function not used, kick :)

File: test/fixtures/onChain.js [hunk 3]
```diff
@@ -105,10 +105,10 @@ export default [
 
   // still poor test
   // TODO: init state with returnData first
-  { code: ['00', OP.RETURNDATASIZE], step: 0 },
+  { code: [OP.STOP, OP.RETURNDATASIZE], step: 1 },
```
Comment (johannbarbie) [inline test/fixtures/onChain.js]: seems to be related to first comment. is it that stop opcode should increase step counter?

Comment (pinkiebell) [reply test/fixtures/onChain.js]: Right :)

File: contracts/mocks/SpendingConditionMock.sol (hunk 2 above)
Comment (pinkiebell) [reply contracts/mocks/SpendingConditionMock.sol]: Kicking 👍

File: contracts/EVMRuntime.sol (hunk 1 above)
Comment (pinkiebell) [reply contracts/EVMRuntime.sol]: > seems to be related to first comment. is it that stop opcode should increase step counter?

👍

**Pull request merged**